env0/utils.go

```diff
@@ -0,0 +1,141 @@
+package env0
+
+import (
+	"errors"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+// https://stackoverflow.com/questions/56616196/how-to-convert-camel-case-string-to-snake-case
+func toSnakeCase(str string) string {
+	snake := matchFirstCap.ReplaceAllString(str, "${1}_${2}")
+	snake = matchAllCap.ReplaceAllString(snake, "${1}_${2}")
+	return strings.ToLower(snake)
+}
+
+func boolPtr(b bool) *bool {
+	return &b
+}
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func intPtr(i int) *int {
+	return &i
+}
+
+// Extracts values from the resourcedata, and writes it to the interface.
+func readResourceData(i interface{}, d *schema.ResourceData) error {
+	val := reflect.ValueOf(i).Elem()
+	for i := 0; i < val.NumField(); i++ {
+		fieldName := val.Type().Field(i).Name
+		// Assumes golang is CamalCase and Terraform is snake_case.
+		fieldNameSC := toSnakeCase(fieldName)
+		field := val.Field(i)
+		fieldType := field.Type()
+
+		dval, ok := d.GetOk(fieldNameSC)
+		if !ok {
+			continue
+		}
+
+		switch fieldType.Kind() {
+		case reflect.Ptr:
+			switch field.Type().Elem().Kind() {
+			case reflect.Int:
+				i := dval.(int)
+				field.Set(reflect.ValueOf(&i))
+			case reflect.Bool:
+				b := dval.(bool)
+				field.Set(reflect.ValueOf(&b))
+			}
+		case reflect.Slice:
+			switch field.Type() {
+			case reflect.TypeOf([]client.ModuleSshKey{}):
```

Comment:
This means that we must implement every possible array type that is supported by the env0 API. Is that correct?

If that's the case, how about throwing on some `default` clause so that every developer that'll use this util will get alerted on development phase that he's trying to parse a non-supported array attribute?

Reply:
Yes. Complex types require special handling. I have some ideas on how to streamline it (to be done in the future).
 But yes I agree. A good compromise for now - alerting developers or returning an error is preferred. Will be fixed.

Reply:
https://github.com/env0/terraform-provider-env0/pull/279/commits/a59b4bbbb4c4f0f861546b4848b2c55e5040adbb